### PR TITLE
Make auto save happen after fade 

### LINKED ISSUE #1410: Make auto-save happen after fade
to prevent the screenshot to be just all black, and the saving message not being visible.

### PR DESCRIPTION
**Pull request by DTanxxx**
Call SaveHelper.AutoSave after transitions instead.

fixes #1410

### CODE REVIEW

**Comment by hhyyrylainen** [comment]
You didn't have to open a new PR. You can force push anything over a PR to completely change the commits in it.

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]
It looks like the autosave in the editor still happens too soon.
Putting a breakpoint in `OnFinishTransitioning` doesn't seem to actually even trigger. It is probably because if auto-evo is ready when entering the editor there is a separate path in addition to the one in _Process how the editor ready gets called.

**Comment by hhyyrylainen** [comment]
I don't think the way the editor entry works after the latest changes is good.
The save needs to happen after the auto-evo results are applied. And now it always makes an autosave once the fade has finished, so it can trigger the auto save while auto-evo is still doing stuff, and showing a loading screen.
The approach before e0767a3 needed just a bit of tweaking to have the right behaviour.

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]
My review is basically this: https://github.com/Revolutionary-Games/Thrive/pull/1590#issuecomment-689453655

File: src/microbe_stage/editor/MicrobeEditor.cs [hunk 1]
```diff
@@ -372,6 +379,16 @@ public void OnFinishLoading(Save save)
...
     }
 
+    public void OnFinishTransitioning()
+    {
+        transitionFinished = true;
+
+        if (!IsLoadedFromSave)
+        {
+            InitEditorFresh();
+        }
```
Comment (hhyyrylainen) [inline src/microbe_stage/editor/MicrobeEditor.cs:389]: I don't think this is the right way either.

File: src/microbe_stage/editor/MicrobeEditor.cs [hunk 2]
```diff
@@ -288,6 +293,8 @@ public override void _Ready()
 
         gui.Init(this);
 
+        transitionFinished = false;
+
```
Comment (hhyyrylainen) [inline src/microbe_stage/editor/MicrobeEditor.cs:297]: Isn't this default by false? Why is this needed?

File: src/microbe_stage/editor/MicrobeEditor.cs [hunk 3]
```diff
@@ -910,6 +930,9 @@ private void InitEditor()
 
     private void InitEditorFresh()
     {
+        if (!transitionFinished)
+            return;
+
```
Comment (hhyyrylainen) [inline src/microbe_stage/editor/MicrobeEditor.cs:935]: Instead of this check, extend the auto-evo check like this:
```
        if (!CurrentGame.GameWorld.IsAutoEvoFinished())
        {
            ready = false;
            LoadingScreen.Instance.Show("Loading Microbe Editor", CurrentGame.GameWorld.GetAutoEvoRun().Status);
        }
        else if (!transitionFinished)
        {
             ready = false;
        }
        else
        {
            OnEditorReady();
        }
```

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]

File: src/microbe_stage/editor/MicrobeEditor.cs (hunk 1 above)
Comment (DTanxxx) [reply src/microbe_stage/editor/MicrobeEditor.cs:389]: When the transition finishes, I thought we can call InitEditorFresh again to actually execute the whole method? Since if either auto evo or transition hasn't finished, ready would still be false and EditorReady is not called.

Comment (hhyyrylainen) [reply src/microbe_stage/editor/MicrobeEditor.cs:389]: That method does other setup as well, which I want to run before the fade-in works.
Look at closely what it does, it does a bunch of setup even when it detects that auto-evo isn't ready.

Comment (DTanxxx) [reply src/microbe_stage/editor/MicrobeEditor.cs:389]: InitEditorFresh is also called by OnEnterEditor which is called in _Ready, so wouldn't the setups in InitEditorFresh run just fine? The setups would run multiple times though, considering how InitEditor is called in OnEnterEditor, OnFinishLoading, and OnFinishTransitioning.

Comment (hhyyrylainen) [reply src/microbe_stage/editor/MicrobeEditor.cs:389]: No if you have the current code where you bail out of `InitEditorFresh` if transition is not finished.
This wasn't even my main comment about this...

https://github.com/Revolutionary-Games/Thrive/pull/1590#discussion_r485507994 is what I actually want, InitEditorFresh to only skip the `OnEditorReady` call if transition is still in progress.

**Review by hhyyrylainen — APPROVED** [review]
I fixed a very serious bug where the editor transition finished was never actually called, because when the transition was started editor was null so nothing got called.
Also changed how the editor init finalizing interacts with the transition finished property to be to my liking.
Could you test if this still works like it should?

File: src/microbe_stage/editor/MicrobeEditor.cs (hunk 1 above)
Comment (DTanxxx) [reply src/microbe_stage/editor/MicrobeEditor.cs]: Oooh ok I see. But what if the fade transition finishes after both calls to InitEditorFresh (since InitEditor is called in two places I assumed that InitEditorFresh would be called twice in total) so that OnEditorReady is never reached in the if statement? Or does that never actually happen?

Comment (hhyyrylainen) [reply src/microbe_stage/editor/MicrobeEditor.cs]: The other place where InitEditor is called, when a save is loaded. It should never be the case that InitEditor is called twice.
The logic is that when loading the editor attempt to call InitEditor, it does some setup always, and if both auto-evo and the fade is ready then it calls editor ready. If editor ready has not been called the _Process in the editor checks if auto-evo and fade are ready now, if they are then on ready is called.

**Comment by hhyyrylainen** [comment]
I'm going to merge this as I noticed 2 unrelated bugs that I need to fix ASAP.

File: src/microbe_stage/editor/MicrobeEditor.cs (hunk 1 above)
Comment (DTanxxx) [reply src/microbe_stage/editor/MicrobeEditor.cs]: Right I see. _Process method takes care of the fade not being finished on time issue.
